src/CrudPanel.php

```diff
@@ -237,16 +239,91 @@ public function sort($items)
      *          App/Models/Address defined by a company() method on the user model and an address() method on the
      *          company model, the 'App/Models/Address' string will be returned.
      *
-     * @param $relationString String Relation string. A dot notation can be used to chain multiple relations.
+     * @param string $relationString Relation string. A dot notation can be used to chain multiple relations.
+     * @param int $length Optionally specify the number of relations to omit from the start of the relation string. If
+     *        the provided length is negative, then that many relations will be omitted from the end of the relation
+     *        string.
+     * @param Model $model Optionally specify a different model than the one in the crud object.
      *
-     * @return string relation model name
+     * @return string Relation model name.
      */
-    private function getRelationModel($relationString)
+    public function getRelationModel($relationString, $length = null, $model = null)
     {
-        $result = array_reduce(explode('.', $relationString), function ($obj, $method) {
+        $relationArray = explode('.', $relationString);
+
+        if (! isset($length)) {
```

Comment:
Actually, if length defaulted to 0...we wouldn't need that `isset` (I suppose - on the basis of if they send in garbage, they get garbage out).

Or we could check if that `$length` really is a number of some sort.

OTOH at some stage we have to assume that developers aren't too silly (which is not sensible when I'm the developer :P).